src/components/leaderboards/List.vue

```diff
@@ -1,52 +1,54 @@
 <script setup lang="ts">
 import { useAsyncState } from '@vueuse/core'
-import { computed, ref } from 'vue'
+import { ref } from 'vue'
 import { Leaderboards } from '../../lib/api/Leaderboards'
+import { StatusFilter } from '../../lib/api/data-contracts'
 
-enum IncludeDeleted {
-	NotDeleted,
-	Deleted,
-	All
-}
-
-const includeDeleted = ref(IncludeDeleted.NotDeleted)
-const search = ref<string>('')
+const limit = 256;
+const status = ref<StatusFilter>('Published')
+const query = ref<string>('')
+const searchedQuery = ref<string>('')
+const searchedStatus = ref<StatusFilter>()
 
 const leaderboardClient = new Leaderboards({
 	baseUrl: import.meta.env.VITE_BACKEND_URL
 })
 
 const {
-	state: allBoards,
+	state: boards,
 	error,
 	isLoading,
 	execute
 } = useAsyncState(async () => {
-	const resp = await leaderboardClient.listLeaderboards({
-		includeDeleted: true
+	const resp = query.value ? await leaderboardClient.searchLeaderboards({
+		q: query.value,
+		status: status.value,
+		limit: limit
+	}) : await leaderboardClient.listLeaderboards({
+		status: status.value,
+		limit: limit
 	})
 
-	return resp.data
+	return resp.data.data
 }, [])
 
-const boardsFiltered = computed(() =>
-	allBoards.value.filter(
-		(board) =>
-			includeDeleted.value === IncludeDeleted.All ||
-			(includeDeleted.value === IncludeDeleted.NotDeleted) ===
-				(board.deletedAt === null)
-	)
-)
-
-const boardsSearched = computed(() =>
-	boardsFiltered.value.filter(
-		(board) =>
-			!search.value ||
-			board.name
-				.toLocaleLowerCase()
-				.includes(search.value.trim().toLocaleLowerCase())
-	)
-)
+function search() {
+	execute()
+	searchedQuery.value = query.value
+	searchedStatus.value = status.value
+}
+
+function clear() {
+	query.value = ''
+	searchedQuery.value = ''
+	execute()
+}
+
+function filterChanged() {
+	if (!query.value){
+		search()
+	}
+}
```

Comment:
How do we get the boards of the same query and of the other status with that `if` check?

Reply:
I think it'll be better that way